Gems/Multiplayer/Code/Source/Components/NetBindComponent.cpp

```diff
@@ -167,10 +170,13 @@ namespace Multiplayer
 
     void NetBindComponent::Deactivate()
     {
-        AZ_Assert(
-            m_needsToBeStopped == false,
-            "Entity (%s) appears to have been improperly deleted. Use MarkForRemoval to correctly clean up a networked entity.",
-            GetEntity() ? GetEntity()->GetName().c_str() : "null");
+        if (bg_AssertNetBindOnDeactrivationWithoutMarkForRemoval)
```

Comment:
I understand the reasoning for this to be a quick bandaid, but the underlying issue is really problematic.. if people start deleting entities improperly at runtime now due to this assert not triggering it is going to cause us even worse issues. I'd make this cvar true by default with a task to go back and fix the underlying issue(s). At the very least we need an alternative solution for the controlled entity removal event, and we need an assert put back for deleting a proxied entity. Entity migration may be safe? I have to do a deeper investigation there.